File: Arrays-And-Strings/Challenge-7.js
const rotateImage = (matrix) => {
  const matrixLength = matrix.length;

  for (let i = 0; i < matrixLength; i++) {
    for (let j = i; j < matrixLength; j++) {
      const temp = matrix[i][j];
      matrix[i][j] = matrix[j][i];
      matrix[j][i] = temp;
    }
  }

  for (let i = 0; i < matrixLength; i++) {
    matrix[i].reverse();
  }
};

rotateImage([
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
]);

File: Arrays-And-Strings/Challenge-7.py
# O(N^2) time complexity and O(1) space

def rotate_image(matrix):
  matrix_length = len(matrix)

  for i in range(matrix_length):
    for j in range(i + 1, matrix_length):
      matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

  for i in range(matrix_length):
    matrix[i].reverse()
  return matrix

matrix = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

print(rotate_image(matrix))
